gdal/frmts/hdf5/hdf5dataset.cpp

```diff
@@ -42,6 +42,10 @@
 #include "hdf5.h"
 #include "hdf5dataset.h"
 
+#if H5_VERS_MAJOR == 1 && H5_VERS_MINOR <= 8 && (H5_VERS_MINOR < 8 || H5_VERS_RELEASE < 9)
```

Comment:
this test is wrong
Should be something like (untested !)

#if H5_VERS_MAJOR > 1 || (H5_VERS_MAJOR == 1 && (H5_VERS_MINOR > 8 ||  (H5_VERS_MINOR == 8 && H5_VERS_RELEASE >=9)))

Reply:
Sorry about the constant tab problems. I have now fixed my Emacs installation to never use tabs whenever I edit my GDAL trees. ;-) They should not reoccur.